### PR TITLE
tests: Clean up InitFrameworkForRayTracingTest

### PR DESCRIPTION
**Pull request by sjfricke**
going to add supports/tests for some Ray Tracing stuff, but the `InitFrameworkForRayTracingTest` function was a little outdated and needed some refactoring and moved this out to its own PR

### CODE REVIEW

**Review by ncesario-lunarg — COMMENTED** [review]
These changes LGTM, but haven't investigated the CI failure yet.

@Tony-LunarG I added you as a reviewer as I _think_ you have some knowledge with this setup, but if I'm mistaken (I might be conflating with GPU-AV setup), feel free to ignore :)

**Comment by Tony-LunarG** [comment]
Fixing InitFrameworkForRayTracingTest to call InitFramework before checking for requested extensions has resulted in several tests being run rather than being skipped.  These tests have validation errors, which causes them to fail.  In the couple I looked at, three things seem to be needed
- The buffer device address feature needs to be enabled
- Buffers used in buffer device address calls need to be created with VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT
- Memory bound to the buffer needs to be allocated with VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT_KHR
The failing tests are:
[  FAILED  ] VkLayerTest.ValidateGetRayTracingCaptureReplayShaderGroupHandlesKHR
[  FAILED  ] VkLayerTest.ValidateCreateAccelerationStructureKHR
[  FAILED  ] VkLayerTest.ValidateCreateAccelerationStructureKHRReplayFeature
[  FAILED  ] VkLayerTest.ValidateCmdTraceRaysKHR
[  FAILED  ] VkLayerTest.ValidateCmdTraceRaysIndirectKHR
[  FAILED  ] VkLayerTest.ValidateCmdBuildAccelerationStructuresKHR
[  FAILED  ] VkLayerTest.ObjInUseCmdBuildAccelerationStructureKHR
[  FAILED  ] VkLayerTest.TestCmdCopyMemoryToAccelerationStructureKHR
I patched a couple of the tests locally and those three things were what was needed to pass, but patching all the tests isn't as good as adding a method to create VkBufferObjs with VkMemoryAllocateFlagsInfo as an input.

**Comment by sjfricke** [comment]
> Fixing InitFrameworkForRayTracingTest to call InitFramework before checking for requested extensions

I moved it back for now and added a TODO note, I don't have a good way to iterate all the test failures

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -2473,45 +2471,44 @@ bool InitFrameworkForRayTracingTest(VkRenderFramework *renderFramework, bool isK
...
-        renderFramework->AddRequiredExtensions(VK_KHR_BUFFER_DEVICE_ADDRESS_EXTENSION_NAME);
-        renderFramework->AddRequiredExtensions(VK_KHR_SPIRV_1_4_EXTENSION_NAME);
-        renderFramework->AddRequiredExtensions(VK_KHR_DEFERRED_HOST_OPERATIONS_EXTENSION_NAME);
-        renderFramework->AddRequiredExtensions(VK_KHR_PIPELINE_LIBRARY_EXTENSION_NAME);
+    framework->AddRequiredExtensions(VK_KHR_GET_MEMORY_REQUIREMENTS_2_EXTENSION_NAME);
+    if (is_khr) {
+        framework->AddRequiredExtensions(VK_KHR_ACCELERATION_STRUCTURE_EXTENSION_NAME);
+        framework->AddRequiredExtensions(VK_KHR_RAY_TRACING_PIPELINE_EXTENSION_NAME);
+        framework->AddRequiredExtensions(VK_KHR_RAY_QUERY_EXTENSION_NAME);
+        framework->AddRequiredExtensions(VK_KHR_GET_MEMORY_REQUIREMENTS_2_EXTENSION_NAME);
```
Comment (Tony-LunarG) [inline tests/layer_validation_tests.cpp:2479]: Memory requirements 2 is already added above

**Comment by Tony-LunarG** [comment]
#4296 filed for skipping tests

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -2473,45 +2471,44 @@ bool InitFrameworkForRayTracingTest(VkRenderFramework *renderFramework, bool isK
...
-               renderFramework->RequiredExtensionsNotSupported().c_str());
+               framework->RequiredExtensionsNotSupported().c_str());
         return false;
     }
 
-    renderFramework->InitFramework(user_data, enabled_features);
+    framework->InitFramework(&framework->Monitor(), enabled_features);
 
-    if (renderFramework->IsPlatform(kMockICD) || renderFramework->DeviceSimulation()) {
+    if (!mockicd_valid && (framework->IsPlatform(kMockICD) || framework->DeviceSimulation())) {
```
[the remark this answers is not in the document]
Comment (sjfricke) [reply tests/layer_validation_tests.cpp:2499]: yes, the reason for this change is I use mockICD to tests some tests I will push after this, but decide to pull the "clean up this function" into its own PR

File: tests/layer_validation_tests.cpp (hunk 1 above)
Comment (sjfricke) [reply tests/layer_validation_tests.cpp]: good catch, removed

**Review by Tony-LunarG — APPROVED** [review]
LGTM - thanks!

**Pull request merged**